src/formState.ts

```diff
@@ -374,6 +389,13 @@ function newListFieldState<T, U>(key: string, rules: Rule<T, U[]>[], config: Obj
       }
     },
 
+    reset() {
+      if (originalCopy) {
+        this.set(originalCopy);
+        this.rows.every((r) => r.reset());
```

Comment:
Nice! It makes sense that in the non-house-of-cards approach that this would work; put the original copies back (of both the entities + the row state wrappers that we cache) and tell each wrapper to reset, which will itself have captured each rows original value.